### DISCUSSION TITLE
Set default theme on V17

### DISCUSSION THREAD

**Discussion by PedroArs99** [keycloak/keycloak#10491] [Archive]
Hi!,

Im migrating from Keycloak 16.1.1 to 17.0.0 but i´m struggling while setting the custom theme. 

 I´m able to import the custom theme to the server but the only way i found to set it as active is logging on the console and set it up manually. I´m looking some way to set it programatically like on v16 with the `KEYCLOAK_DEFAULT_THEME` env variable.

Is there something similar on v17?

**Comment by semangard** [comment]
cf. issue https://github.com/keycloak/keycloak/issues/10874

**Comment by maveonair** [comment]
If you start your server with `kc.sh start --spi-theme-default=custom-theme`, then it should use the supplied custom theme, according to: https://github.com/keycloak/keycloak/blob/bfce612641a70e106b20b136431f0e4046b5c37f/services/src/main/java/org/keycloak/theme/DefaultThemeManager.java#L57

This worked for me with Keycloak 18.

Also note that there is a setting that allows you to set the default theme for the welcome page: `--spi-theme-welcome-theme=custom-theme`.

**Comment by pedroigor** [comment]
@maveonair @PedroArs99 I'm wondering if it makes sense to provide explicit options for configuring themes? The reason is that it should be a quite common configuration.

@stianst @DGuhr Wdyt?

I know we don't have the extension story properly addressed, but a `theme` namespace in configuration should not change once we improve this area.
